data_schema/convert_value.py

```diff
@@ -154,6 +154,11 @@ def _convert_value(self, value, format_str):
         if self.is_string(value):
             value = datetime.strptime(value, format_str) if format_str else parse(value)
 
+        # It is assumed that value is a datetime here. If it isn't a datetime, then it is a bad value like
+        # a number that is too large to be parsed as an integer
+        if type(value) != datetime:
+            raise InvalidDateFormatException('Invalid date format')
```

Comment:
can you also include the bad value in the error message?

Reply:
yeah, i added a test for it as well